mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -1012,11 +1017,40 @@ private void CleanupFiles (string dest, HashSet<string> goodfiles)
 						continue;
 					}
 
-					if (string.IsNullOrWhiteSpace (PreserveTag)) { // only do this if there was no -preserve
+					Action actuallyDelete = () => {
 						string newname = typefile.FullName + ".remove";
-						try { System.IO.File.Delete(newname); } catch (Exception) { }
-						try { typefile.MoveTo(newname); } catch (Exception) { }
-						Console.WriteLine("Class no longer present; file renamed: " + Path.Combine(nsdir.Name, typefile.Name));
+						try { System.IO.File.Delete (newname); } catch (Exception ex) { Warning ("Unable to delete existing file: {0}", newname); }
+						try { typefile.MoveTo (newname); } catch (Exception ex) { Warning ("Unable to rename to: {0}", newname); }
+						Console.WriteLine ("Class no longer present; file renamed: " + Path.Combine (nsdir.Name, typefile.Name));
+					};
+
+					if (string.IsNullOrWhiteSpace (PreserveTag)) { // only do this if there was not a -preserve
+						using (TextWriter writer = OpenWrite (typefile.FullName, FileMode.Truncate))
```

Comment:
I know this one is a bit confusing, but It's actually to persist changes made to the XML document in the call to `UpdateAssemblyVersions(e, assembly, GetAssemblyVersions(assemblyName), false)` several lines above. This method doesn't actually persist those changes to disk, and in most other cases where a type is actually in the assembly, it gets persisted later on ... but in this code path when a type has actually been removed from the assembly it never ends up saving it. And so I needed that to be persisted because that function removes the "classic" reference ... then later on when we do the "unified" assembly, we can see that the type isn't tagged with "classic", and it's safe to remove if it's not in the unified assembly either.